src/oak_d.py

```diff
@@ -397,10 +399,20 @@ async def get_images(
         # Accumulator for timestamp calculation later
         seconds_float: float = None
 
-        if COLOR_SENSOR in self.sensors:
-            captured_data = await cls.worker.get_color_image()
-            arr, captured_at = captured_data.np_array, captured_data.captured_at
+        # We really want to do this as in-sync as possible to reduce latency between color & depth
```

Comment:
^

Reply:
Are there any relevant docs about image syncing?

Reply:
https://docs.luxonis.com/projects/api/en/latest/tutorials/message_syncing/#software-syncing

I have in my unstaged changes on this branch, a draft of integrating the above solution in this code. I know it's not best practice to commit a bunch of changes to main after a review, but if curious/ready, we can review that on this PR too.

Reply:
It works— I tested using the rgb-d-overlay cam

Reply:
https://www.loom.com/share/304bf698a0d642e6b009aade42d7f8ab?sid=a63947d0-ae8f-4e3a-8dfa-2d12e1028ddc

Reply:
I added the changes that implement syncing